bin/targetselection.py

```diff
@@ -45,54 +36,51 @@ def main():
         
     #- Loop over bricks collecting target selection flags
     #- and targets that passed the cuts
-    targetflags = list()
-    targets = list()
     t0 = time()
-    nbrick = 0
-    if ns.numproc > 1:
-        bnames, bfiles, results = \
-            map_tractor(_select_targets_brickfile, ns.src, \
-                bricklist=bricklist, numproc=ns.numproc)
-        #- unpack list of tuples into tuple of lists
-        targets, targetflags = zip(*results)
-    else:
-        for brickname, filename in iter_tractor(ns.src):
-            if (bricklist is not None) and (brickname not in bricklist):
-                continue
-            
-            nbrick += 1
-            xtargets, xflags = _select_targets_brickfile(filename)
-            targets.append(xtargets)
-            targetflags.append(xflags)
 
-            if ns.verbose and nbrick%50 == 0:
-                rate = nbrick / (time() - t0)
-                print('{} bricks; {:.1f} bricks/sec'.format(nbrick, rate))
+    def _select_targets_brickfile(filename):
+        '''Wrapper function for performing target selection on a single brick file
+        
+        Used by _map_tractor() for parallel processing'''
+        objects = read_tractor(filename)
+        targetflag = select_targets(objects)
+        keep = (targetflag != 0)
+
+        objects = objects[keep]
+        targetflag = targetflag[keep]
+        numobs = calc_numobs(objects, targetflag)
+
+        targets = desitarget.targets.finalize(objects, targetflag, numobs)
+
+        return fix_tractor_dr1_dtype(targets)
+
+    nbrick = np.zeros((), dtype='i8')
+    targets = [] 
+
+    def collect_results(result):
+        ''' wrapper function for the critical reduction operation,
+            that occurs on the main parallel process '''
+        if ns.verbose and nbrick%50 == 0:
+            rate = nbrick / (time() - t0)
+            print('{} bricks; {:.1f} bricks/sec'.format(nbrick, rate))
+        nbrick[...] += 1
+        targets.append(result)
```

Comment:
I added this commit as a second thought, mostly to demonstrate another way of building 'targets' list.(which is out-of-order)
The commit before this was still using the result of map_tractor (which is in-order).

I can revert this commit.